boa/src/builtins/number/mod.rs

```diff
@@ -318,45 +318,53 @@ impl Number {
     ///  - [ECMAScript reference][spec]
     ///  - [MDN documentation][mdn]
     ///
-    /// [spec]: https://tc39.es/ecma262/#sec-number.prototype.toexponential
+    /// [spec]: https://tc39.es/ecma262/#sec-number.prototype.toprecision
     /// [mdn]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/toPrecision
     #[allow(clippy::wrong_self_convention)]
     pub(crate) fn to_precision(
         this: &Value,
         args: &[Value],
         context: &mut Context,
     ) -> Result<Value> {
-        let precision_arg = args.get(0);
+        let precision_var = args.get(0).cloned().unwrap_or_default();
 
+        // 1 & 6
         let mut this_num = Self::this_number_value(this, context)?;
-        if precision_arg.is_none() || !this_num.is_finite() {
+        // 2 & 4
+        if precision_var == Value::undefined() || !this_num.is_finite() {
             return Self::to_string(this, &[], context);
         }
 
-        let precision = match precision_arg.unwrap().to_integer(context)? as i32 {
-            x @ 1..=100 => x as usize,
+        // 3
+        let precision = match precision_var.to_integer_or_infinity(context).unwrap() {
```

Comment:
Replace the `.unwrap()` with `?`, this way it matches the spec's `? ToIntegerOrInfinity(precision)`

As an example `(3).toPrecision(Symbol.match)` will result in a panic and should instead throw the error.